File: srv/www/test_cgi.py
#!/usr/bin/python3
import cgi
import os

print("Content-Type: text/html\r\n\r\n")
print("<html><body>")
print("<h2>Python CGI Test</h2>")
print("<p>Method: {}</p>".format(os.environ.get("REQUEST_METHOD", "")))
if os.environ.get("REQUEST_METHOD") == "POST":
    form = cgi.FieldStorage()
    for key in form.keys():
        print("<p>{}: {}</p>".format(key, form[key].value))

print("</body></html>")

File: srv/www/test_cgi.php
#!/usr/bin/php-cgi
<?php
echo "Content-Type: text/html\r\n\r\n";
echo "<html><body>";
echo "<h2>PHP CGI Test</h2>";
echo "<p>Method: " . $_SERVER["REQUEST_METHOD"] . "</p>";
if ($_SERVER["REQUEST_METHOD"] == "POST") {
    foreach ($_POST as $key => $value) {
        echo "<p>$key = $value</p>";
    }
}
echo "</body></html>";
?>
